types/types.go

```diff
@@ -256,6 +257,17 @@ type ImageSource interface {
 	// The Digest field is guaranteed to be provided; Size may be -1.
 	// WARNING: The list may contain duplicates, and they are semantically relevant.
 	LayerInfosForCopy(ctx context.Context, instanceDigest *digest.Digest) ([]BlobInfo, error)
+	// GetDeltaManifest returns the delta manifest for the current image, as well as its type, if it exist.
+	// No error is returned if no delta manifest exists, just a nil slice
+	// It may use a remote (= slow) service.
+	// If instanceDigest is not nil, it contains a digest of the specific manifest instance to retrieve deltas for (when the primary manifest is a manifest list);
+	// this never happens if the primary manifest is not a manifest list (e.g. if the source never returns manifest lists).
+	GetDeltaManifest(ctx context.Context, instanceDigest *digest.Digest) ([]byte, string, error)
```

Comment:
Hum, an API break.

I suppose we’ll now have to figure out how to extend these interfaces without breaking API, it had to come one day.

(Let’s figure out the right approach first and deal with the API later, just highlighting that this will have to happen.)